### PR TITLE
C51-115: Activity Feed Bulk Actions UI + basic setup

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR the bulk actions design and basic functionality has been implemented for activity Feed.

Note: Actions in the dropdown is yet to be implemented.

## How it looks
after

## Technical Details
1. In `ActivityFeed.js`, as new attribute `showBulkActions` is added. And only when this is true, bulk action will be shown. This is required to hide the Bulk Actions in the Dashboard page.
Also instead of calling the `getcount` endpoint, another `get` endpoint is used. This fetches all the Activities(ID Only), so this serves both the count and also all activity ids required for bulk actions.

2. In `ActivityFeed.js`, functions related to Bulk Actions has been copied from `CaseListTable.js`.

3. In `ActivityFilters.html`, the markup for the Bulk Action enabler and dropdown has been implemented.

4. In `ActivityFeed.html`, the bulk action checkboxes has been implemented beside the activity card.

5. In `CaseDetails--placeholder.html` and `FileList.html`, placeholder related markup is updated after implementing the Checkboxes beside the Activity Card.

### CODE REVIEW

File: ang/civicase/ActivityFeed.html [hunk 1]
```diff
@@ -9,22 +19,39 @@
...
+                role="checkbox"
+                tabindex="0"
+                aria-labelledby="Select/Unselect"
+                ng-click="toggleSelected(activity); $event.stopPropagation();">
+                <i ng-if="activity.selected || allActivitiesSelected" class="civicase__checkbox--checked material-icons">check_box</i>
+              </span>
+              <div
+                ng-class="{'civicase__activity-feed__list__vertical_bar': $index< activityGroup.activities.length - 1 }"
+                class="civicase__activity-feed__list__activity-item">
+                <span ng-if="activity.source_contact_id" contact-card avatar="true" contacts="activity.source_contact_id"></span>
```
Comment (swastikpareek) [inline ang/civicase/ActivityFeed.html]: Can we split it into multiple lines like every where else 
```html
<span 
  ng-if="activity.source_contact_id" 
  contact-card avatar="true" 
  contacts="activity.source_contact_id"
 ></span>
```

File: ang/civicase/CaseDetails--placeholder.html [hunk 2]
```diff
@@ -194,16 +194,18 @@ <h3><div class="civicase__loading-placeholder__oneline" style="width: 12em;"></d
...
             <div class="civicase__activity-feed__list">
-              <div class="civicase__activity-feed__activity-container">
-                <div class="civicase__activity-card civicase__activity-card--long" ng-repeat="i in [0,1,2,3,4]">
-                  <div class="panel panel-default">
-                    <div class="panel-body">
-                      <div class="civicase__activity-card-row" style="margin-bottom: 8px;">
-                        <div class="civicase__loading-placeholder__oneline" style="width: 0.8em; font-size: 24px; margin-right: 10px;"></div>
-                        <div class="civicase__loading-placeholder__oneline" style="width: 8em; font-size: 18px;"></div>
-                        <div class="civicase__activity__right-container civicase__loading-placeholder__oneline" style="width: 7em;"></div>
+              <div class="civicase__activity-feed__list__activity-item">
```
Comment (swastikpareek) [inline ang/civicase/CaseDetails--placeholder.html]: Shouldn't this be `civicase__activity-feed__list-item` as it is already inside the activity feed so no need to mention activity again.

File: ang/civicase/CaseDetails--placeholder.html [hunk 3]
```diff
@@ -239,16 +241,18 @@ <h3><div class="civicase__loading-placeholder__oneline" style="width: 12em;"></d
...
     <div>
       <div class="civicase__activity-feed__list">
-        <div class="civicase__activity-card civicase__activity-card--long" ng-repeat="i in [0,1,2]">
-          <div class="panel panel-default">
-            <div class="panel-body">
-              <div class="civicase__activity-card-row" style="margin-bottom: 8px;">
-                <div class="civicase__loading-placeholder__oneline" style="width: 0.8em; font-size: 24px; margin-right: 10px;"></div>
-                <div class="civicase__loading-placeholder__oneline" style="width: 8em; font-size: 18px;"></div>
-                <div class="civicase__activity__right-container civicase__loading-placeholder__oneline" style="width: 7em;"></div>
+        <div class="civicase__activity-feed__list__activity-item">
```
Comment (swastikpareek) [inline ang/civicase/CaseDetails--placeholder.html]: Same here.

File: scss/components/_activity-feed.scss [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
   .civicase__panel-transparent-header {
     margin: auto;
-    width: 830px;
+    width: 870px;
```
Comment (swastikpareek) [inline scss/components/_activity-feed.scss]: Can you please remind me what this change do ? Thx in advance.

**Comment by swastikpareek** [comment]
@deb1990 : There is a small bug in the bulk actions which I noticed. 
When we select some activities from bulk actions and change the filter selection criterion, then the bulk actions selections gets gone from the UI but the message box still shows the selected Activities.
c51-115

So ideally whenever we change any filter the bulk selections should get reset.

Also a small point . Since we have added bulk actions box to the activity list, we should add some UI loading placeholder too for it 
c51-115 - loading missing

I will continue the review after we fixed these issues.

File: scss/components/_activity-feed.scss (hunk 4 above)
Comment (deb1990) [reply scss/components/_activity-feed.scss]: Yes, now that we introduced the Activity Bulk Action Checkbox, the width of the panel holding the activity card and the checkbox needed to be increased.

**Comment by swastikpareek** [comment]
@deb1990 : I still see the issue with select everything bulk actions. After clicking on `Everything` I can't deselect a single checkbox. Please refer this
c51-115 - all activities issue.

Also, If you see the `border-bottom` for case messages is set to `0` for `.civicase__case-list .civicase__bulkactions-message .alert` (was intentionally added for case list) but we will need bottom border here for activities feed.
Can you please add that for activity feed?

**Pull request merged**